feature_engine/feature_selection.py

```diff
@@ -88,3 +89,138 @@ def transform(self, X):
         return X
 
 
+class SelectFeatures(TransformerMixin, BaseEstimator):
+    """
+    SelectFeatures()
+
+    Drops constant and quasi constant variables from a dataframe.
+    Constant variables are those variables which show only one value for all the observations.
+    By default, all constant variables are dropped.
+    Quasi-constant variables are those variables which are almost constant. In other words, those features have the same
+    value for large subset of total observations.
+    This transformer works for both, numerical and categorical variables.
+
+    Parameters
+    ----------
+
+    drop_quasi_constant_features : bool, default=False
+        Whether to drop quasi constant features
+
+    quasi_constant_threshold: float, default=None
+        When drop_quasi_constant_features is True, features showing quasi_constant_threshold percentage of values
+        will be dropped
+
+    Attributes
+    ----------
+
+    constant_features_: list
+        List of constant features in a dataframe
+
+    quasi_constant_features_: list
+        List of quasi constant features in a dataframe
+
+    selected_features_: list
+        List of selected features which are non-constant and non-quasi constant
+
+    """
+    def __init__(self, drop_quasi_constant_features=False, quasi_constant_threshold=None):
+
+        self.drop_quasi_constant_features = drop_quasi_constant_features
+        self.quasi_constant_threshold = quasi_constant_threshold
+
+        self.constant_features_ = []
+        self.selected_features_ = []
+        self.quasi_constant_features_ = []
+        self._input_shape_ = ()
+
```

Comment:
these attributes should go in fit(). I am also thinking that we could have only **selected_features_**, **features_to_remove_**  and **input_shape_** (please note input shape should not have a leading _).

This will simplify the code quite a bit and make the transformer run faster

Reply:
Not sure what do you mean by ```features_to_remove_```? Is it the same as ```variables``` in the below code? If not could you please elaborate?

Reply:
ignore that attribute. Let's just go with `selected_features_`